### PR TITLE
Clarify pytest_ignore_collect docs

### LINKED ISSUE #12383: `pytest_ignore_collect` documentation is misleading
I'm reporting this as a bug because I've hit this multiple times in the wild in a variety of packages implementing `pytest_ignore_collect` in `conftest.py`.

The documentation for `pytest_ignore_collect` states:

> Return True to prevent considering this path for collection.
>
> This hook is consulted for all files and directories prior to calling more specific hooks.
>
> Stops at first non-None result, see [firstresult: stop at first non-None result](https://pytest.org/en/latest/how-to/writing_hook_functions.html#firstresult).
(https://pytest.org/en/latest/reference/reference.html#std-hook-pytest_ignore_collect)

While this isn't incorrect, it's misleading. A lot of implementations I've seen assume they're supposed to return `False` for paths that are not to be ignored. In fact, I didn't realize that returning `None` was the right thing to do, unless I've found this bit in the changelog:

> If after updating to this version you see that your norecursedirs setting is not being respected, it means that a conftest or a plugin you use has a bad pytest_ignore_collect implementation. Most likely, your hook returns False for paths it does not want to ignore, which ends the processing and doesn’t allow other plugins, including pytest itself, to ignore the path. The fix is to return None instead of False for paths your hook doesn’t want to ignore.
(https://pytest.org/en/latest/changelog.html#id141)

Could you please improve the documentation for `pytest_ignore_collect` to make it clear that normally `True` or `None` should be returned? This is especially problematic since returning `False` makes `--ignore` silently ignored.

### PR DESCRIPTION
**Pull request by nicoddemus**
Fixes #12383

### CODE REVIEW

File: src/_pytest/hookspec.py [hunk 1]
```diff
@@ -311,7 +311,12 @@ def pytest_collection_finish(session: "Session") -> None:
 def pytest_ignore_collect(
     collection_path: Path, path: "LEGACY_PATH", config: "Config"
 ) -> Optional[bool]:
-    """Return True to prevent considering this path for collection.
+    """Return ``True`` to prevent considering this path for collection.
```
Comment (bluetech) [inline src/_pytest/hookspec.py:314]:
```suggestion
    """Return ``True`` to ignore this path for collection.
```

File: src/_pytest/hookspec.py [hunk 2]
```diff
@@ -311,7 +311,12 @@ def pytest_collection_finish(session: "Session") -> None:
...
+
+    Return ``None`` to let other plugins ignore the path for collection.
+
+    Returning ``False`` will forcefully consider this path to be considered
+    for collection, without giving a chance of other plugins to ignore this file.
```
Comment (bluetech) [inline src/_pytest/hookspec.py:319]:
```suggestion
    Returning ``False`` will forcefully *not* ignore this path for collection,
    without giving a chance for other plugins to ignore this file.
```

**Review by bluetech — APPROVED** [review]
I understand why you wrote it the way you did, but "consider this path to be considered for collection" sounded a bit clumsy to me... I put a suggestion to lean solely on the "ignore" terminology. But either way is OK by me.

**Comment by nicoddemus** [comment]
> I understand why you wrote it the way you did, but "consider this path to be considered for collection" sounded a bit clumsy to me... I put a suggestion to lean solely on the "ignore" terminology. But either way is OK by me.

Oh no I appreciate very much the suggestions -- I could not give this too much thought TBH, wanted to get the ball rolling and hoped a good review would improve my initial wording. So success! Hehehe.

**Pull request merged**